File: plugins/unit_test/tests/test_sample.php
<?php
// Sample PHP unit-style test file

function assertEqual($a, $b, $name) {
    if ($a === $b) {
        echo "PASS: $name\n";
    } else {
        echo "FAIL: $name — expected " . var_export($b, true) . " got " . var_export($a, true) . "\n";
    }
}

assertEqual(1 + 1,              2,       "addition");
assertEqual(strtoupper("hello"), "HELLO", "strtoupper");
assertEqual(strlen("Qt6"),       3,       "strlen");
assertEqual(implode(",", ["a","b","c"]), "a,b,c", "implode");
assertEqual(10 % 3,             1,       "modulo");

// Intentional failure
assertEqual(42, 99, "intentional_failure");

File: plugins/unit_test/tests/test_sample.pl
#!/usr/bin/perl
use strict;
use warnings;

sub assert_eq {
    my ($got, $expected, $name) = @_;
    if ($got eq $expected) {
        print "PASS: $name\n";
    } else {
        print "FAIL: $name — expected '$expected' got '$got'\n";
    }
}

assert_eq(1 + 1,            "2",     "addition");
assert_eq(uc("hello"),      "HELLO", "uc");
assert_eq(length("Qt6"),    "3",     "length");
assert_eq(join(",", "a","b","c"), "a,b,c", "join");

# Intentional failure
assert_eq("foo", "bar", "intentional_failure");
